wolfssl/wolfcrypt/types.h

```diff
@@ -135,6 +135,44 @@ decouple library dependencies with standard string, memory and so on.
         #endif
     #endif
 
+    #ifndef HAVE_EMPTY_AGGREGATES
+        /* The C standards don't define empty aggregates, but gcc and clang do.
+         * We need to accommodate them for one of the same reasons C++ does --
+         * templates that conditionally result in empty aggregates, e.g. in
+         * hash.h.
+         */
+        #if !defined(WOLF_C89) && defined(__GNUC__) &&  \
+                !defined(__STRICT_ANSI__) &&            \
+                HAVE_ANONYMOUS_INLINE_AGGREGATES + 0 == 1
+            #define HAVE_EMPTY_AGGREGATES 1
+        #else
+            #define HAVE_EMPTY_AGGREGATES 0
+        #endif
+    #endif
+
+    #define _WOLF_AGG_DUMMY_MEMBER_HELPER2(a, b, c) a ## b ## c
+    #define _WOLF_AGG_DUMMY_MEMBER_HELPER(a, b, c) _WOLF_AGG_DUMMY_MEMBER_HELPER2(a, b, c)
+    #if HAVE_EMPTY_AGGREGATES + 0 == 1
```

Comment:
This should just be `ifdef HAVE_EMPTY_AGGREGATES`. No need to have weird plus 0 equals one syntax. Then above don't define HAVE_EMPTY_AGGREGATES to 0. It's either defined or not which follows our convention elsewhere.

Reply:
Good idea.  It was a mistake to introduce this competing convention with `HAVE_ANONYMOUS_INLINE_AGGREGATES` (6a92db7722).  I've refactored both of them, so that undefined signifies false, and defined to `1` to signifies true.  That's the convention with autotools `HAVE_*` variables (c.f. generated `config.h`), and it's consistent with wolfssl convention.

Post-refactor, users can still supply override-on and override-off values for either of them as before, with `0` forcing off and `1` forcing on, but now they can also force it on with an empty value, consistent with wolfssl convention, fixing semantics for `user_settings.h`.

Note that on the compiler command line, `-Dfoo` with no explicit value automagically becomes a definition to `1`.  Moreover, invoking a preprocessor macro with an empty arg list automagically acts like passing a `0`.  There be dragons lurking in the preprocessor...